site/static/main.ts

```diff
@@ -2,6 +2,9 @@
 
 declare const BASEURL, hljs;
 
+// IIFE to prevent function declarations from moving into the global scope
```

Comment:
The `.d.ts` structure mimics the file structure of the project (see how the types field in the package mimics the main field). There is no built in tool to bundle `.d.ts` files, nor is there a reason to bundle them, since they only get used in the compilation/editor.

Re: static site ts changes. I honestly would prefer not to change the static site assets, but I can't help it in the current build setup. The site ts is a script, not a module. This is fine, and ts does support mixed compilations. The problem is that with declarations on, that script file has ambient declarations made for it; so everything which leaks into the global scope gets a declaration in the resulting `.d.ts`. Usually, I wouldn't care... Nobody's planning on using that `.d.ts` file, anyway, except it has a declaration visibility error which arises in the compilation. The type of the first argument to `renderExample`  is considered private, which causes a declaration generation error. The solution I chose was to simply remove the global leaks by doing all the work inside an IIFE. IMO, a slightly more complete/better fix would be a separate tsconfig and a separate build for the site and the library (so the site can be compiled with declarations off, rather than on), but this works if that's too complex.